integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -179,6 +182,93 @@ object DataLoadProcessBuilderOnSpark {
     updateLoadStatus(model, partialSuccessAccum)
   }
 
+  def insertDataUsingGlobalSortWithInternalRow(sparkSession: SparkSession,
+      scanResultRDD : RDD[InternalRow],
+      model: CarbonLoadModel,
+      hadoopConf: Configuration): Array[(String, (LoadMetadataDetails, ExecutionErrors))] = {
+    val originRDD = scanResultRDD
+
+    val sc = sparkSession.sparkContext
+    val modelBroadcast = sc.broadcast(model)
+    val partialSuccessAccum = sc.longAccumulator("Partial Success Accumulator")
+    val sortStepRowCounter = sc.longAccumulator("Sort Processor Accumulator")
+    val writeStepRowCounter = sc.longAccumulator("Write Processor Accumulator")
+
+    hadoopConf
+      .set(CarbonCommonConstants.CARBON_WRITTEN_BY_APPNAME, sparkSession.sparkContext.appName)
+
+    val conf = SparkSQLUtil.broadCastHadoopConf(sc, hadoopConf)
+    val configuration = DataLoadProcessBuilder.createConfiguration(model)
+
+    // 1. Convert internalRow to object array
+    val fields = Util
+      .convertToSparkSchemaFromColumnSchema(model.getCarbonDataLoadSchema.getCarbonTable, true)
+      .fields
+      .toSeq
+
+    val dataTypes = fields.map(field => field.dataType)
+    val map: mutable.Map[String, GenericDataType[_]] = mutable.Map[String, GenericDataType[_]]()
+    CommonUtil.convertComplexDataType(map, configuration)
+    val rdd = originRDD.map { internalRow =>
+      CommonUtil.getObjectArrayFromInternalRowAndConvertComplexTypeForGlobalSort(internalRow,
+        fields,
+        map)
+    }
+
+    // 2. sort
+    var numPartitions = CarbonDataProcessorUtil.getGlobalSortPartitions(
+      configuration.getDataLoadProperty(CarbonCommonConstants.LOAD_GLOBAL_SORT_PARTITIONS))
+    if (numPartitions <= 0) {
+      numPartitions = originRDD.partitions.length
+    }
+    // Because if the number of partitions greater than 1, there will be action operator
+    // (sample) in
+    // sortBy operator. So here we cache the rdd to avoid do input and convert again.
+    if (numPartitions > 1) {
+      rdd.persist(StorageLevel.fromString(
+        CarbonProperties.getInstance().getGlobalSortRddStorageLevel()))
+    }
+    val sortColumnsLength = model.getCarbonDataLoadSchema.getCarbonTable.getSortColumns.size()
+    val sortColumnDataTypes = dataTypes.take(sortColumnsLength)
+    val rowComparator = GlobalSortHelper.generateRowComparator(sortColumnDataTypes)
+    val sortRDD = rdd.sortBy(x => getKey(x, sortColumnsLength, sortColumnDataTypes),
```

Comment:
move `getKey(x, sortColumnsLength, sortColumnDataTypes)` to next line
use pattern like ```
rdd.sortBy { row =>
  ...
}
```

Reply:
ok